toolkit/components/processtools/ProcInfo_haiku.cpp

```diff
@@ -0,0 +1,99 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "mozilla/ProcInfo.h"
+#include "mozilla/ProcInfo_linux.h"
+#include "mozilla/Sprintf.h"
+#include "mozilla/Logging.h"
+#include "mozilla/ScopeExit.h"
+#include "mozilla/ipc/GeckoChildProcessHost.h"
+#include "nsMemoryReporterManager.h"
+#include "nsWhitespaceTokenizer.h"
+
+#include <cstdio>
+#include <cstring>
+#include <unistd.h>
+#include <dirent.h>
+
+#include <kernel/OS.h>
+
+#define NANOPERSEC 1000000000.
+
+namespace mozilla {
+
+int GetCycleTimeFrequencyMHz() { return 0; }
+
+nsresult GetCpuTimeSinceProcessStartInMs(uint64_t* aResult) {
+  const thread_id thread = find_thread(NULL);
+  thread_info info;
+  if (B_OK != get_thread_info(thread, &info)) {
+    return NS_ERROR_FAILURE;
+  }
+  const bigtime_t microseconds = info.user_time + info.kernel_time;
+  *aResult = microseconds / 1000;
```

Comment:
`Maybe should use get_team_usage_info() like processcontroll er

https://github.com/haiku/haiku/blob/master/src/apps/processcontroller/TeamBarMenuItem.cpp#L188